keydata.go

```diff
@@ -404,6 +419,30 @@ func (d *KeyData) snapModelAuthKey(auxKey AuxiliaryKey) ([]byte, error) {
 	return hmacKey, nil
 }
 
+func (d *KeyData) snapModelAuthKey(auxKey AuxiliaryKey) ([]byte, error) {
+	kdf := d.data.AuthorizedSnapModels.kdf
+	if kdf == nil {
+		return d.snapModelAuthKeyLegacy(auxKey)
+	}
+	if !kdf.Alg.Available() {
+		return nil, errors.New("invalid KDF digest algorithm")
+	}
+
+	alg := d.data.AuthorizedSnapModels.alg
+	if alg.Hash == crypto.Hash(0) {
+		return nil, errors.New("invalid digest algorithm")
+	}
```

Comment:
is there some constraints between alg.Size() and kdf.AlgSize() ?  the code below anyway seems to use the same algo for both

Reply:
> The size of the key we derive should be at least the size of the output of the digest used for the HMAC

shouldn't this bit be checked then?